### PR TITLE
Add new profile model

### PR DESCRIPTION
**Pull request by daoneandonly**
# Change Profile model
Hi team, I've changed the core database structure of the app so please read the following notes and model carefully.
We will now use just one model for a user which is called `Profile`. The `Profile` data has everything we need of a single user. If anything more comes up, we can add this of course.
The model looks as follows;

```js
{
    "_id": {
        "$oid": "60632c4f2d65f7ed68f39dff"
    },
    "email": "test@gmail.com",
    "password": "$2b$10$R/SEUma2JyWQmpfcJBf7zOgdbDc5AnzCvDybHqBKouqIR1o3Jgl6y",
    "profileData": {
        "firstName": "Test",
        "LastName": "Testy",
        "age": 18,
        "bio": "bio",
        "photo": "/img/61667b1cdf12cc4fb17283b1a617cca8.jpg",
        "gender": "he",
        "countries": ["country 1", "country 2", "country 3"],
        "preferences": {
            "gender": "he",
            "minAge": 18,
            "maxAge": 69,
            "maxDistance": 200
        }
    }
}
```

## A few notes 
* ✏We will have to use `findById()` and `findByIdAndUpdate()` to select a user. `findOne` and related functions are deprecated and can no longer be used.
* ❌ Don't use eslint exception lines if a rule doesn't suit you. Either disable the rule entirely or just ignore it. It's there to help, not to add clutter to the code.
* 📜 Every instance where we need user data, the function for it is added. It's gonna be needed on almost every page (to check for log in later as well).
* 🌍 We can use a seperate database collection to keep track of all the Countries, that is still in place and will have to be updated.
* 👤 There is a minimal setup in place for keeping the user logged in, using ⚠ **hardcoded** ⚠data of an `id`. We'll have to use some way to get that `id` from the user. A good thing to look into could be [cookie-session-middleware](http://expressjs.com/en/resources/middleware/cookie-session.html),
* 🏄 The current flow is _still_ a little janky: You register, then you log in again, and then get asked for additional information. A lot of up front information. There is also no actual homescreen. Maybe reconsider in what order things are asked for. Do we need a `name` at login? Making pages should be a piece of cake by now, so don't be scared to add things or break it up in more steps.
* 👩‍🎨 Styling could still be better. If anything it should be more consistent throughout the app.
* 📸 Multer is completely not working. This just **has** to be functional.
* 🟢 You can log in with these credentials: `test@gmail.com` / `1234`

## Todo
* [x] Updated the Profile model that uses all information.
* [ ] Setup to keep track of a logged in user. (Preferably track this in the header, i.e. change login/register buttons with "Hello [name]")
* [ ] Remove the hardcoded `id` in the functions.
* [ ] Update flow. Have home screen etc.
* [ ] Make styling consistent.
* [ ] Make image upload completely functional.

TLDR: There is still quite a bit of work left. Pick your task, consult google, and don't be shy to ask questions.

### CODE REVIEW

**Review by maxwieringa — APPROVED** [review]
I think the model looks great. I'm gonna try to fix multer, I don't know why it doesn't work on the heroku app but I'll look into it. We should merge this as soon as possible so I can fix my feature and make it work with the new and improved model.
